src/osd/PG.cc

```diff
@@ -3752,11 +3753,18 @@ int PG::build_scrub_map_chunk(
     return ret;
   }
 
-
-  get_pgbackend()->be_scan_list(map, ls, deep, seed, handle);
+  get_pgbackend()->be_scan_list(map, ls, seed, handle);
   _scan_rollback_obs(rollback_obs, handle);
   _scan_snaps(map);
 
+  if (deep) {
+    // XXX: Drop the PG lock because we are just reading the data and 
```

Comment:
Oh, and you can configure more than 1 thread/shard, but it's still worth ensuring that each item can be processed in a bounded amount of time.